packages/mobile-native-mcp-server/src/tools/plan/sfmobile-native-user-input-triage/tool.ts

```diff
@@ -49,35 +49,39 @@ export class SFMobileNativeUserInputTriageTool extends AbstractWorkflowTool<
       **User Input:**
       ${JSON.stringify(input.userUtterance)}
 
-      ## Required Analysis
+      ## ⚠️ CRITICAL REQUIREMENT - DO NOT PROCEED WITHOUT REQUIRED FIELDS ⚠️
 
-      You must provide a structured response that extracts as much relevant information as possible from the user input. Use intelligent inference where appropriate, but clearly document your assumptions.
+      **YOU MUST NOT PROCEED TO THE NEXT STEP UNTIL ALL REQUIRED FIELDS ARE COLLECTED.**
 
-      ### 1. Extract Core Properties
+      ALL required fields specified in the provided JSON schema MUST be present.
 
-      Look for the following information in the user input:
+      If ANY required fields are missing after analyzing the user input:
+      
+      1. STOP IMMEDIATELY
+      2. DO NOT proceed to the next step in the workflow
+      3. DO NOT populate the schema with empty or placeholder values
+      4. EXPLICITLY ask the user for EACH missing required field
+      5. EXPLAIN why this information is necessary
+      6. WAIT for the user's response with the missing information
 
-      - **Platform**: Is this for iOS, Android, or both? If not specified, make a reasonable assumption based on context.
-      - **Project Name**: Extract or infer a suitable project name from the app description.
-      - **Package Name**: Generate a proper package identifier (e.g., com.company.appname).
-      - **Organization**: Extract or infer the organization/company name.
+      ## Analysis Guidelines
 
-      ### 2. Extract Salesforce Configuration (if mentioned)
+      Extract as much relevant information as possible from the user input:
 
-      Look for any Salesforce-specific requirements:
+      1. **Core Properties**:
+         - Extract platform, project name, package name, and organization
+         - For project name and package name, use information from the app description
+         - Generate a proper package identifier format (e.g., com.company.appname)
 
-      - **Connected App Client ID**: If the user mentions OAuth or Connected App configuration.
-      - **Connected App Callback URI**: If OAuth flows are discussed.
-      - **Login Host**: If specific Salesforce environments are mentioned (login.salesforce.com, test.salesforce.com, custom domains).
+      2. **Salesforce Configuration**:
+         - Connected App Client ID and Callback URI are REQUIRED and cannot be assumed
```

Comment:
We can try this out as ad hoc guidance here, but we're going to want to codify/generalize the concept of required vs. optional vs. inferred fields in the form of structured schema, and backing guidance.

Reply:
my hope was that it would do that already just by inspecting the requiredness of the schema but seems like it needs some extra nudging